storage/retrieval.go

```diff
@@ -133,69 +153,81 @@ func (b *Bucket) getLogChunks(ctx context.Context, buildID string) ([]LogChunkIn
 		}
 	}
 
-	if err := iter.Err(); err != nil {
-		return nil, nil, errors.Wrap(err, "getting log chunks")
-	}
-
 	sortLogChunksByStartTime(buildChunks)
 	sortLogChunksByStartTime(testChunks)
 
 	return buildChunks, testChunks, nil
 }
 
-// filterLogChunksByTestID returns (1) the resulting slice of log chunks after
-// filtering for chunks with the given test ID and (2) the appropriate test
-// execution time range for the given test ID.
-func filterLogChunksByTestID(chunks []LogChunkInfo, testID string) ([]LogChunkInfo, TimeRange) {
+// parseTestIDs parses test IDs from the buildKeys that correspond to test metadata files
+// and sorts them by creation time.
+func parseTestIDs(buildKeys []string) ([]model.TestID, error) {
+	var testIDs []model.TestID
+	for _, key := range buildKeys {
+		if !strings.HasSuffix(key, metadataFilename) {
+			continue
+		}
+		if !strings.Contains(key, "/tests/") {
+			continue
+		}
+		testID, err := testIDFromKey(key)
+		if err != nil {
+			return nil, errors.Wrap(err, "getting test ID from metadata key")
+		}
+		testIDs = append(testIDs, model.TestID(testID))
+	}
+
+	sort.Slice(testIDs, func(i, j int) bool {
+		return testIDs[i].Timestamp().Before(testIDs[j].Timestamp())
+	})
+
+	return testIDs, nil
+}
+
+// testExecutionWindow returns the TimeRange from the creation of this test to the creation
+// of the next test. If the testID isn't found the returned TimeRange is unbounded.
```

Comment:
Should it be an error if testID isn't found?

Reply:
Like based on my reading of the usage that should never happen.

Reply:
The old logic had to find a TestChunk within the testid and that could fail but I don't think we can ever just not have the test metadata in the list.

Reply:
got it. 
I fixed it to differentiate between the case where the testID isn't provided (like when we're fetching all the build's logs) and when a non-existent ID is provided.